learning_wavelets/models/learnlet_layers.py

```diff
@@ -250,7 +286,8 @@ def call(self, analysis_coeffs):
                 wav_norm = self.wav_filters_norm[i_scale]
                 detail = detail * wav_norm
             if self.res:
-                image = self.convs_groupping[i_scale](detail) + image
+                image = self.convs_groupping_fixed[i_scale](detail[...,0:self.n_shearlets]) + image
+                image = self.convs_groupping_train[i_scale](detail[...,self.n_shearlets:self.n_tiling]) + image
```

Comment:
```suggestion
                image = self.convs_groupping_fixed[i_scale](detail[..., :self.n_shearlets]) + image
                image = self.convs_groupping_train[i_scale](detail[...,self.n_shearlets:]) + image
```

We had mentioned this bug actually.

Here the problem is that the total number of details is `n_shearlets + n_tiling`.
Let's do a quick example, imagine `detail = [0, 1, 2, 3, 4, 5]`, `n_shearlets = 2` and `n_tiling = 4`.
Then we have:

```python
details[0:n_shearlets] == details[0:2] == [0, 1]
details[n_shearlets:n_tiling] = details[2:4] = [2, 3]
```

This also means that you don't need to store `n_tiling` in this class.

Reply:
this has been discussed